test/functional/internal/test_utils.py

```diff
@@ -15,61 +15,75 @@
+from dynamodb_encryption_sdk.transform import dict_to_ddb
 
-from ..functional_test_vector_generators import attribute_test_vectors
+from ..functional_test_utils import diverse_item
 
 
-def get_test_item(standard_dict_format, partition_key, sort_key):
-    attributes = attribute_test_vectors("serialize")
+def get_test_item(standard_dict_format, partition_key, sort_key=None):
+    attributes = diverse_item()
 
     attributes = {"attr_" + str(pos): attribute[0] for pos, attribute in enumerate(attributes)}
-    attributes["partition-key"] = {"S": partition_key}
-    if sort_key:
-        attributes["sort-key"] = {"S": sort_key}
+    attributes["partition-key"] = partition_key
+    if sort_key is not None:
+        attributes["sort-key"] = sort_key
 
-    if standard_dict_format:
-        attributes = ddb_to_dict(attributes)
+    if not standard_dict_format:
+        attributes = dict_to_ddb(attributes)
     return attributes
 
 
-def get_test_items(standard_dict_format, table_name="table"):
-    items = [
-        get_test_item(standard_dict_format, partition_key="key-1", sort_key=None),
-        get_test_item(standard_dict_format, partition_key="key-2", sort_key=None),
-        get_test_item(standard_dict_format, partition_key="key-3", sort_key="sort-1"),
-        get_test_item(standard_dict_format, partition_key="key-4", sort_key="sort-1"),
-        get_test_item(standard_dict_format, partition_key="key-4", sort_key="sort-2"),
-    ]
+def get_test_items(standard_dict_format, table_name="table", with_sort_keys=False):
+
+    if with_sort_keys:
+        items = [
+            get_test_item(standard_dict_format, partition_key="key-1", sort_key="sort-1"),
+            get_test_item(standard_dict_format, partition_key="key-2", sort_key="sort-1"),
+            get_test_item(standard_dict_format, partition_key="key-2", sort_key="sort-2"),
+        ]
+    else:
+        items = [
+            get_test_item(standard_dict_format, partition_key="key-1"),
+            get_test_item(standard_dict_format, partition_key="key-2"),
+        ]
 
     for pos, item in enumerate(items):
         item["encrypt-me"] = table_name + str(pos)
 
     return {table_name: [{"PutRequest": {"Item": item}} for item in items]}
 
 
-def get_dummy_crypto_config(partition_key_name, sort_key_name, encrypted_attribute_name):
+def get_dummy_crypto_config(partition_key_name=None, sort_key_name=None, sign_keys=False):
     context = EncryptionContext(partition_key_name=partition_key_name, sort_key_name=sort_key_name)
     actions = AttributeActions(
         default_action=CryptoAction.DO_NOTHING,
-        attribute_actions={encrypted_attribute_name: CryptoAction.ENCRYPT_AND_SIGN},
+        attribute_actions={
+            "encrypt-me": CryptoAction.ENCRYPT_AND_SIGN
+        },
     )
-    materials = CryptographicMaterialsProvider()
+    if sign_keys:
+        actions.attribute_actions['partition-key'] = CryptoAction.SIGN_ONLY
+        if sort_key_name is not None:
```

Comment:
NP: You could simplify this by just setting both. If there are override instructions in an `AttributeActions` instance that reference attribute names that are not found in the item, then they are simply never referenced.

Reply:
👍will do!